### PR TITLE
Update accordion-panel state after parent component re-renders #patch

### PR DESCRIPTION
**Pull request by cfree**
## Merge Checklist

[TENT-2367](https://jira.sendgrid.net/browse/TENT-2367). I was getting a bug where my accordion wouldn't render as open. The props passed in were `open: true` but the state of `<AccordionPanelSFC />` was still `false`. Through the profiler, I saw it was rendering twice - the first time as was being displayed (closed), the second time I was getting the props I needed but the state was not updating. I added logic in `componentDidUpdate` to display the new props if they differed from the old ones. This resulted in the accordion opening as expected.

Additionally, this update is to pass the `open` and `onClick` props down through to the `<AccordionTable />` component.

- [x] Includes link to associated Github Issue or JIRA ticket
- [x] Has descriptive PR title and body
- [x] All changes are backwards compatible
- [ ] QA'd by the SendGrid Style Guide designer
- [ ] Code reviewed by a SendGrid frontend lead

### CODE REVIEW

File: src/accordion/accordion-panel.tsx [hunk 1]
```diff
@@ -69,6 +69,12 @@ export class AccordionPanel extends React.Component<
     open: openProps(this.props),
   };
 
+  public componentDidUpdate(prevProps: AccordionPanelProps) {
+    if (prevProps.open !== this.props.open) {
+      this.setState({ open: this.props.open });
```
Comment (chrischalstrom) [inline src/accordion/accordion-panel.tsx]: usually we should not set state from props - it is an anti pattern that gives state multiple ways of being controlled.  but in this situation i think it is okay.

Comment (mmissey) [reply src/accordion/accordion-panel.tsx]: Yeah I agree. I don't really like it, but I don't like `getDerivedStateFromProps` either

Comment (cfree) [reply src/accordion/accordion-panel.tsx]: Agreed. This component was already setting state from props upon mounting the component, but never updated if the props changed after the initial render.

**Comment by thinkingserious** [comment]
Hello @cfree,<br /><br/>Thanks again for the PR!<br />                                              <br />We appreciate your contribution and look forward to                                              continued collaboration. Thanks!                                             <br /><br />Team SendGrid DX

File: src/accordion/accordion.module.scss [hunk 2]
```diff
@@ -247,6 +247,10 @@
     border-top: 1px solid #e9ecef;
   }
 
+  h3 {
+    color: $slate-80;
```
Comment (rossedfort) [inline src/accordion/accordion.module.scss]: sleighty

**Pull request merged**